platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -118,18 +135,49 @@ public String get(String key) {
     }, Clock.SYSTEM));
   }
 
-  public synchronized static void initMetricsRegistry(Map<String, String> defaultTags) {
-    initMetricsRegistry(DEFAULT_METRICS_REPORTERS, DEFAULT_METRICS_PREFIX,
-        METRICS_REPORTER_CONSOLE_REPORT_INTERVAL_DEFAULT, defaultTags);
+  private static void initTestingMetricsReporter() {
+    LOGGER.info("Initializing the testing metric reporter.");
+
+    METER_REGISTRY.add(new SimpleMeterRegistry());
   }
 
-  public synchronized static void initMetricsRegistry(final List<String> reporters,
-      final String prefix, final int reportIntervalSec, Map<String, String> tags) {
+  public synchronized static void initMetricsRegistry(String serviceName, Config config) {
     if (isInit) {
       return;
     }
 
-    metricsPrefix = prefix;
+    List<String> reporters;
+    if (config.hasPath(METRICS_REPORTER_NAMES_CONFIG_KEY)) {
+      reporters = Arrays.asList(config.getString(METRICS_REPORTER_NAMES_CONFIG_KEY).split(","));
+    } else {
+      reporters = DEFAULT_METRICS_REPORTERS;
+    }
+
+    metricsPrefix = DEFAULT_METRICS_PREFIX;
+    if (config.hasPath(METRICS_REPORTER_PREFIX_CONFIG_KEY)) {
+      metricsPrefix = config.getString(METRICS_REPORTER_PREFIX_CONFIG_KEY);
+    }
+
+    int reportIntervalSec = DEFAULT_METRIC_REPORT_INTERVAL_SEC;
+    if (config.hasPath(METRICS_REPORT_INTERVAL_CONFIG_KEY)) {
+      reportIntervalSec = config.getInt(METRICS_REPORT_INTERVAL_CONFIG_KEY);
+    }
+
+    Map<String, String> defaultTags = new HashMap<>();
+
+    // Add the service name and other given tags to the default tags list.
+    defaultTags.put("app", serviceName);
```

Comment:
what happens when there are multiple instances of a service ?

Reply:
They all get the same `app` tag, which is what we want. The instance details are added at the pod level if it's running in k8s. We could add the host name or IP also if needed later.

Reply:
Forgot that scrape config  will determine those additional labels like pod name that will get injected

Reply:
Sorry, what is this scrape config? Are the pod/node etc tags automatically added to each metric somehow?

Reply:
@avinashkolluru the way metrics are sent to Prom is this:

1. service exposes an endpoint `/metrics` where metrics are spit in Prom compatible fashion. No tags until now but with micro-meter, service itself will be adding some additional tags.
2. Prom collector calls the `/metrics` endpoint of each pod periodically (where is this configured??) and adds the labels of the pod to all those metrics before pushing to Prometheus. This is how we have `app` label on all metrics as of today, though the service isn't setting any labels.

With the new library, service code also can add labels, like `tenantid`